src/github.com/matrix-org/dendrite/clientapi/routing/routing.go

```diff
@@ -93,11 +94,16 @@ func Setup(
 			return writers.SendEvent(req, device, vars["roomID"], vars["eventType"], vars["txnID"], nil, cfg, queryAPI, producer)
 		}),
 	)
-	r0mux.Handle("/rooms/{roomID}/state/{eventType}",
+	r0mux.Handle("/rooms/{roomID}/state/{eventType:[a-z._-]+(?:\\/)?}",
```

Comment:
I don't think you need the group. I'm pretty sure you can write this as:

```go
"{eventType:[a-z._-]+/?}"
```

Hmm, this regexp is a tad more restrictive that what synapse does. I'm not sure it that is a problem though. Think the synapse regexp is:

```go
"{event_type:[^/]+/?}"
```

based on https://github.com/matrix-org/synapse/blob/master/synapse/rest/client/v1/room.py#L190